### PR TITLE
Reorganize the crate and rename to bindgen.

### LINKED ISSUE #21: merge into upstream!
Did you consider to merge your modifications into my repo? We both added a lot of features, it would be great if we could join forces.

### LINKED ISSUE #398: Publish the executable on crates.io
Since this is a useful library that is used in also many non-servo projects, I think it will be convinient to have it in crates.io.  Then also we can do `cargo install servo-rust-bindgen` .

### PR DESCRIPTION
**Pull request by emilio**
Fixes #398 
Fixes #21 

r? @fitzgen

### CODE REVIEW

**Comment by emilio** [comment]
cc @Yamakaky

**Comment by Yamakaky** [comment]
You should change the version to 0.20, so that it's considered an upgrade on crates.io.

**Comment by emilio** [comment]
Good catch @Yamakaky :)

File: README.md [hunk 1]
```diff
@@ -117,20 +117,20 @@ In `Cargo.toml`:
 build = "build.rs"
 
 [build-dependencies]
-libbindgen = "0.1"
+bindgen = "0.1"
```
Comment (fitzgen) [inline README.md]: This should probably suggest "0.20" now right?

**Comment by fitzgen** [comment]
I'll need to update my tutorial now :)

**Comment by emilio** [comment]
@bors-servo r=fitzgen,Yamakaky

Yay!

**Comment by fitzgen** [comment]
@emilio did you publish a new version to crates.io?

**Pull request merged**